includes/Feed/FeedFileUploadHandler.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Feed;
+
+use SkyVerge\WooCommerce\PluginFramework\v5_10_0 as Framework;
+
+
+defined( 'ABSPATH' ) || exit;
+/**
+ *  A class responsible for responding to Facebook feed file requests.
+ *
+ * @since 2.6.0
+ */
+class FeedFileUploadHandler {
+
+	/**
+	 * Feed file handling and manipulation class.
+	 *
+	 * @var FeedFileHandler
+	 */
+	protected $feed_file_handler;
+
+
+	/**
+	 * Singleton instance.
+	 *
+	 * @var FeedFileUploadHandler
+	 */
+	protected static $instance;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param FeedFileHandler $feed_file_handler Feed file handling.
+	 */
+	public function __construct( $feed_file_handler ) {
+		$this->feed_file_handler = $feed_file_handler;
+		add_action( 'woocommerce_api_' . $feed_file_handler::REQUEST_FEED_ACTION, array( $this, 'handle_feed_request' ) );
+	}
+
+	/**
+	 * Initialize the class.
+	 *
+	 * @since 2.6.0
+	 * @param FeedFileHandler $feed_file_handler Feed file handling.
+	 * @return void
+	 */
+	public static function init( $feed_file_handler ) {
+		if ( null === self::$instance ) {
+			self::$instance = new self( $feed_file_handler );
+		}
```

Comment:
Why are we making this class a singleton? I think it would be better to instantiate this class in our existing `facebook_for_woocommerce()` plugin singleton.

Reply:
Done.